ConcernsCaseWork/ConcernsCaseWork/Pages/Shared/_ValidationErrors.cshtml

```diff
@@ -1,16 +1,23 @@
 @using ConcernsCaseWork.Models;
-@using Microsoft.AspNetCore.Mvc.ModelBinding
+@using Microsoft.AspNetCore.Mvc.ModelBinding;
+@using ConcernsCaseWork.Enums;
 @model ConcernsCaseWork.Pages.Base.AbstractPageModel
 
+
 @if (!Model.ModelState.IsValid)
 {
+	var unsortedErrors = Model.ModelState
+				.Where(s => s.Value?.ValidationState == ModelValidationState.Invalid)
+				.Select(e => new { e.Key, Msg = string.Join("<br />", e.Value.Errors.Select(errs => errs?.ErrorMessage)) });
+
+	//if page model doesn't have sort enum then leave default order. if it does have sort enum order accordingly
+	var sortedError = !Model.ErrorSortOrder.HasValue ? unsortedErrors : Model.ErrorSortOrder.Value.Equals(SortOrder.Ascending) ? unsortedErrors.OrderBy(e => e.Key) : unsortedErrors.OrderByDescending(e => e.Key);
```

Comment:
SUGGESTION:

	var errors = Model.ModelState
				.Where(s => s.Value?.ValidationState == ModelValidationState.Invalid)
				.Select(e => new { e.Key, Msg = string.Join("<br />", e.Value.Errors.Select(errs => errs?.ErrorMessage)) });
				
				if (Model.ErrorSortOrder.HasValue)
				{
					errors = Model.ErrorSortOrder.Value.Equals(SortOrder.Ascending) ? errors .OrderBy(e => e.Key) : errors .OrderByDescending(e => e.Key);
				}

I think this might be slightly easier to read

Reply:
Feel free to challenge